app/services/analytics.ts

```diff
@@ -0,0 +1,74 @@
+import { action } from '@ember-decorators/object';
+import Service from '@ember/service';
+import Ember from 'ember';
+import config from 'ember-get-config';
+
+export default class Analytics extends Service {
+    metrics = Ember.inject.service();
+    session = Ember.inject.service();
+
+    @action
+    click(category, label, extraInfo) {
+        let extra = extraInfo;
+        if (extra && typeof extra !== 'string') {
+            // This is to remove the event object when used with onclick
+            extra = null;
+        }
+        Ember.get(this, 'metrics')
+            .trackEvent({
+                category,
+                action: 'click',
+                label,
+                extra,
+            });
+
+        return true;
+    }
+
+    track(category, actionName, label, extraInfo) {
+        let extra = extraInfo;
+        if (extra && typeof extra !== 'string') {
+            extra = null;
+        }
+        Ember.get(this, 'metrics')
+            .trackEvent({
+                category,
+                actionName,
+                label,
+                extra,
+            });
+        return true;
+    }
+
+    trackPage(page, title, publicPrivate, resourceType) {
```

Comment:
We could add types and default values to enforce what these can be. Maybe like:
```ts
trackPage (
    page: string,
    title: string,
    publicPrivate: 'public' | 'private' | 'n/a' = 'n/a',
    resourceType: string = 'n/a',
) { ...
```
Leaning on TS like that (and getting to the point where we can make TS errors fail the build) will let us simplify how we approach a lot of things.

Reply:
(also maybe could be cleaner to replace `publicPrivate` with an optional boolean, maybe, but I was more excited about types)